codegen/src/test/kotlin/software/amazon/smithy/rust/codegen/smithy/BaselineTransformTest.kt

```diff
@@ -0,0 +1,75 @@
+/*
+ * Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package software.amazon.smithy.rust.codegen.smithy
+
+import io.kotest.matchers.shouldBe
+import org.junit.jupiter.api.Test
+import software.amazon.smithy.model.Model
+import software.amazon.smithy.model.shapes.OperationShape
+import software.amazon.smithy.model.shapes.ShapeId
+import software.amazon.smithy.model.shapes.StructureShape
+import software.amazon.smithy.model.traits.MixinTrait
+import software.amazon.smithy.rust.codegen.testutil.asSmithyModel
+import software.amazon.smithy.rust.codegen.util.cloneOperation
+import software.amazon.smithy.rust.codegen.util.orNull
+import software.amazon.smithy.rust.codegen.util.rename
+import software.amazon.smithy.rust.codegen.testutil.generatePluginContext
+import software.amazon.smithy.rust.codegen.smithy.customize.CombinedCodegenDecorator
+import software.amazon.smithy.rust.codegen.smithy.customize.RequiredCustomizations
+import software.amazon.smithy.rust.codegen.smithy.customize.NoOpEventStreamSigningDecorator
+import software.amazon.smithy.rust.codegen.smithy.customizations.ClientCustomizations
+import software.amazon.smithy.rust.codegen.smithy.generators.client.FluentClientDecorator
+import software.amazon.smithy.rust.codegen.smithy.CodegenVisitor
+
+fun String.asSmithyModel_Version2(sourceLocation: String? = null): Model {
+    val processed = letIf(!this.startsWith("\$version")) { "\$version: \"2.0\"\n$it" }
+    return Model.assembler().discoverModels().addUnparsedModel(sourceLocation ?: "test.smithy", processed).assemble()
+        .unwrap()
+}
+
+class BaselineTransformTest {
```

Comment:
This should probably be called `CodegenVisitorTest`, with the test case referring to `baselineTransform`.